test_communication/test/test_messages_c.cpp

```diff
@@ -1101,67 +1102,41 @@ void get_message(test_msgs__msg__BoundedSequences * msg, size_t msg_num)
   }
 }
 
+#define EXPECT_SEQUENCES_EQ(pred, a, b) \
+  do { \
+    EXPECT_EQ(a.size, b.size) << "Sequences have different length"; \
+    for (size_t i = 0; i < std::min(a.size, b.size); ++i) { \
+      pred(a.data[i], b.data[i]) << "Sequences differ at element " << i; \
+    } \
+  } while (0)
+
 DEFINE_FINI_MESSAGE(test_msgs__msg__BoundedSequences)
 template<>
 void verify_message(test_msgs__msg__BoundedSequences & message, size_t msg_num)
 {
   test_msgs__msg__BoundedSequences expected_msg;
   get_message(&expected_msg, msg_num);
-  for (size_t i = 0; i < expected_msg.bool_values.size; ++i) {
-    EXPECT_EQ(expected_msg.bool_values.data[i],
-      message.bool_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.byte_values.size; ++i) {
-    EXPECT_EQ(expected_msg.byte_values.data[i],
-      message.byte_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.char_values.size; ++i) {
-    EXPECT_EQ(expected_msg.char_values.data[i],
-      message.char_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.float32_values.size; ++i) {
-    EXPECT_FLOAT_EQ(expected_msg.float32_values.data[i],
-      message.float32_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.float64_values.size; ++i) {
-    EXPECT_DOUBLE_EQ(expected_msg.float64_values.data[i],
-      message.float64_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.int8_values.size; ++i) {
-    EXPECT_EQ(expected_msg.int8_values.data[i],
-      message.int8_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.uint8_values.size; ++i) {
-    EXPECT_EQ(expected_msg.uint8_values.data[i],
-      message.uint8_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.int16_values.size; ++i) {
-    EXPECT_EQ(expected_msg.int16_values.data[i],
-      message.int16_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.uint16_values.size; ++i) {
-    EXPECT_EQ(expected_msg.uint16_values.data[i],
-      message.uint16_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.int32_values.size; ++i) {
-    EXPECT_EQ(expected_msg.int32_values.data[i],
-      message.int32_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.uint32_values.size; ++i) {
-    EXPECT_EQ(expected_msg.uint32_values.data[i],
-      message.uint32_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.int64_values.size; ++i) {
-    EXPECT_EQ(expected_msg.int64_values.data[i],
-      message.int64_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.uint64_values.size; ++i) {
-    EXPECT_EQ(expected_msg.uint64_values.data[i],
-      message.uint64_values.data[i]);
-  }
-  for (size_t i = 0; i < expected_msg.string_values.size; ++i) {
-    EXPECT_EQ(0, strcmp(message.string_values.data[i].data,
-      expected_msg.string_values.data[i].data));
+
+  EXPECT_SEQUENCES_EQ(EXPECT_EQ, expected_msg.bool_values, message.bool_values);
```

Comment:
@rotu nit: syntax looks a bit awkward, can we add another set of macros so that it reads like:

```c++
EXPECT_SEQUENCE_EQ(/* ... */);
EXPECT_FLOAT_SEQUENCE_EQ(/* ... */);
EXPECT_DOUBLE_SEQUENCE_EQ(/* ... */);
```
?

Reply:
There are already too many macros. I felt dirty enough making a new one (vs a template, so the stack trace would be correct), but I would prefer to keep it as one.

Reply:
> There are already too many macros. 

That's true, but at the same time I still find `EXPECT_SEQUENCES_EQ(EXPECT_EQ, ...)` a bit terse. That could easily be personal bias. @ivanpauno @dirk-thomas @shane what do you think? If you're OK with it, I'll approve.

Reply:
I prefer @hidmic style, but I don't mind strongly.

Reply:
I'm happy with it as is. I do like the sound of singular `EXPECT_SEQUENCE_EQ` since I read it as one sequence is expected to be equal to another.

If it was being added to a header to become a test utility, I would recommend an interface where `EXPECT_EQ` was the default if used like `EXPECT_SEQUENCE_EQ(seq1, seq2)`, but could be overridden with a third argument like `EXPECT_SEQUENCE_EQ(seq1, seq2, EXPECT_DOUBLE_EQ)`. However [it looks like it would be a lot more complex](https://stackoverflow.com/questions/3046889/optional-parameters-with-c-macros) than is needed for a macro used in a single file